src/core/agent/services/connectionService.ts

```diff
@@ -137,10 +138,39 @@ class ConnectionService extends AgentService {
       );
       const operation = await this.agent.modules.signify.resolveOobi(url);
       const connectionId = operation.response.i;
-      await this.createConnectionKeriMetadata(connectionId, {
+      const connectionMetadata: any = {
         alias: operation.alias,
         oobi: url,
-      });
+      };
+      if (url.includes("groupId")) {
+        const params = new URLSearchParams(new URL(url).search);
+        const groupId = params.get("groupId") as string;
+        const identifierWithGroupId =
+          await AriesAgent.agent.identifiers.getKeriIdentifierByGroupId(
+            groupId
+          );
+        if (identifierWithGroupId) {
+          connectionMetadata.groupId = groupId;
+        } else {
+          //TODO: Open a pop up to get the display name and colors inputs
+          const displayName = utils.uuid();
+          const colorGenerator = new ColorGenerator();
+          const newColor = colorGenerator.generateNextColor();
+
+          await AriesAgent.agent.identifiers.createIdentifier({
+            displayName,
+            method: IdentifierType.KERI,
+            colors: [newColor[1], newColor[0]],
+            theme: 0,
+            groupMetadata: {
+              groupId: utils.uuid(),
+              groupInitiator: false,
+              groupCreated: false,
+            },
+          });
+        }
+      }
+      await this.createConnectionKeriMetadata(connectionId, connectionMetadata);
```

Comment:
Shouldn't we be storing the groupId now if it was set?

Reply:
Here is the snippet above this line
```
if (identifierWithGroupId) {
          connectionMetadata.groupId = groupId;
        }
```

Reply:
Yes, but what about the else block where we call `await AriesAgent.agent.identifiers.createIdentifier`? We have `utils.uuid()`